src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -1,140 +1,393 @@
 package com.endava.internship.collections;
 
+
 import java.util.Collection;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Objects;
+import java.util.Arrays;
 import java.util.ListIterator;
 
+
 public class StudentList implements List<Student> {
+    private int capacity;
+    private int size;
+    private Student[] array;
+
+
+    {
+        size = 0;
+    }
+    public StudentList() {
+        this.capacity = 10;
+        this.array = new Student[this.capacity];
+    }
+
+    public StudentList(int initialCapacity) {
+        if (initialCapacity >= 0) {
+            this.capacity = initialCapacity;
+            this.array = new Student[this.capacity];
+        } else {
+            if (initialCapacity < 0) {
+                throw new IllegalArgumentException("Initial capacity must be greater than 0");
+            }
+
+            this.capacity = 10;
+            this.array = new Student[this.capacity];
+        }
+
+    }
+
+    public StudentList(Collection<? extends Student> collection) {
+        if (!Objects.nonNull(collection)) {
```

Comment:
to simplify reading, can be used `Objects.isNull()`

Reply:
Fixed!